Apps/Src/Temperature.c

```diff
@@ -230,7 +230,9 @@ ErrorStatus Temperature_UpdateSingleChannel(uint8_t channel){
 #ifndef SIMULATION
 		rawTemperatures[board][channel][medianFilterIdx] = milliVoltToCelsius(Minions[board].aux.a_codes[0] / 10);
 #else
-		rawTemperatures[board][channel][medianFilterIdx] = Simulator_getTemperature(board * MAX_TEMP_SENSORS_PER_MINION_BOARD + channel);
+		if (board * MAX_TEMP_SENSORS_PER_MINION_BOARD + channel < NUM_TEMPERATURE_SENSORS) {
+			rawTemperatures[board][channel][medianFilterIdx] = Simulator_getTemperature(board * MAX_TEMP_SENSORS_PER_MINION_BOARD + channel);
+		}
 #endif
```

Comment:
probably want to return error with an else

Reply:
the above statement `rawTemperatures[board][channel][medianFilterIdx] = milliVoltToCelsius(Minions[board].aux.a_codes[0] / 10);` i thought just filled all the temps up 64 regardless, so it would be an array {temps, repeated 62 times, 0, 0} or something. Also I think our simulator does tell user if we don't have 62 temperature values in our JSON file. if i did an error here though would it be to go into the fault state?

Reply:
We should probably fix the bug in the non-simulator code, since we don't check the array bounds there either